tests/FSharp.Compiler.ComponentTests/Language/AnonymousRecordsTests.fs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace FSharp.Compiler.ComponentTests.Language
+
+open Xunit
+open FSharp.Test.Compiler
+
+module ``Anonymous records tests`` =
+
+    [<Fact>]
+    let ``Simple with- should remove one field`` () =
+        Fsx """
+type A = { X: int; Y: int }
+let a = { X = 1; Y = 2 }
+let b = {| a with- Y |}
+        """
+         |> compile
+         |> shouldSucceed
+    
+         [<Fact>]
+    let ``with- shouldn't remove the only field`` () =
+        Fsx """
+type A = { X: int }
+let a = { X = 1 }
+let b = {| a with- X |}
```

Comment:
This would be quite a use case for https://github.com/dotnet/fsharp/issues/6941

Reply:
I would expect a `with-` to always remove the field. When the fields already present are controlled elsewhere, I would want this to be a guarantee.

Reply:
The flow of fields should allow an empty state where more fields can be added later.